src/libcrun/cgroup.c

```diff
@@ -142,7 +143,38 @@ libcrun_cgroup_is_container_paused (struct libcrun_cgroup_status *status, bool *
 
   ret = read_all_file (path, &content, NULL, err);
   if (UNLIKELY (ret < 0))
-    return ret;
+    {
+      errno = crun_error_get_errno (err);
+      /* If the file is missing and we were checking for freezer.state
+         (so either cgroup v1 or hybrid), it may be the freezer is
+         simply disabled. In such case the container cannot be paused.
+         On cgroup v2 freezer is always there.
+      */
+      if (errno != ENOENT || cgroup_mode == CGROUP_MODE_UNIFIED)
+        return ret;
+
+      /* Even with freezer disabled, its directory is still there. But
+         when it's disabled it has type tmpfs, while on systems with
+         freezer enabled, its type is cgroupfs. Use that to determine
+         whether freezer is enabled or not.
+      */
+      struct statfs freezer_stat;
+      if (statfs (CGROUP_ROOT "/freezer", &freezer_stat))
+        return crun_make_error (err, errno, "error when using statfs on `%s`", CGROUP_ROOT "/freezer");
```

Comment:
we need to release the previous `err` before return

Reply:
Done. I wasn't sure whether I need to do that, and somehow I forgot to ask